src/kixi/comms/components/kafka.clj

```diff
@@ -140,50 +140,63 @@
 
 (defn create-consumer
   [handle-result-fn process-msg?-fn kill-chan group-id topics broker-list handler]
-  (let [timeout 100
-        cc {:bootstrap.servers       broker-list
-            :group.id                group-id
-            :auto.offset.reset       :earliest
-            :enable.auto.commit      false
-            :auto.commit.interval.ms timeout}
-        listener (callbacks/consumer-rebalance-listener
-                  (fn [tps]
-                    (info "topic partitions assigned:" tps))
-                  (fn [tps]
-                    (info "topic partitions revoked:" tps)))
-        co (cd/make-default-consumer-options
-            {:rebalance-listener-callback listener})
-        consumer (consumer/make-consumer
-                  cc
-                  (deserializers/string-deserializer)
-                  (deserializers/string-deserializer))
-        _ (cp/subscribe-to-partitions! consumer (vals topics))]
-    (async/go-loop []
-      (let [[val port] (async/alts! [(async/thread
-                                       (cp/poll! consumer {:poll-timeout-ms timeout}))
-                                     kill-chan])]
-        (when-not (= port kill-chan)
-          (doseq [raw-msg (into [] val)]
-            (when-let [msg (some-> raw-msg
-                                   :value
-                                   transit->clj)]
-              (when (process-msg?-fn msg)
-                (async/<! (async/thread
-                            (try
-                              (handle-result-fn (handler msg))
-                              (catch Exception e
-                                (error e (str "Consumer exception processing msg. Raw: " (:value raw-msg) ". Demarshalled: " msg))))))
-                (cp/commit-offsets-async! consumer {(select-keys raw-msg [:topic :partition])
-                                                    {:offset (inc (:offset raw-msg))
-                                                     :metadata (str "Consumer stopping - "(java.util.Date.))}})))))
-        (if-not (= port kill-chan)
-          (recur)
-          (do
-            (cp/clear-subscriptions! consumer)
-            (.close consumer)))))))
+  (async/thread-call
```

Comment:
It's a blocking call (on poll!) so don't want to use 'go' blocks, will jam up the pool. It seems that 'thread' might do some messing around with the code it is given (unsure tbh, many experiments). Using thread-call means nothing can mess with what I give it!